crates/bevy_reflect/src/impls/std.rs

```diff
@@ -217,6 +218,133 @@ impl<T: FromReflect> FromReflect for Vec<T> {
     }
 }
 
+impl<T: FromReflect> Array for VecDeque<T> {
+    #[inline]
+    fn get(&self, index: usize) -> Option<&dyn Reflect> {
+        VecDeque::get(self, index).map(|value| value as &dyn Reflect)
+    }
+
+    #[inline]
+    fn get_mut(&mut self, index: usize) -> Option<&mut dyn Reflect> {
+        VecDeque::get_mut(self, index).map(|value| value as &mut dyn Reflect)
+    }
+
+    #[inline]
+    fn len(&self) -> usize {
+        VecDeque::len(self)
+    }
+
+    #[inline]
+    fn iter(&self) -> ArrayIter {
+        ArrayIter {
+            array: self,
+            index: 0,
+        }
+    }
+}
+
+impl<T: FromReflect> List for VecDeque<T> {
+    fn push(&mut self, value: Box<dyn Reflect>) {
```

Comment:
I wonder if `List` should define a `pop` method. Then we can use sorta use this structure as intended.

Reply:
Agreed, a `pop()` method would make sense.
Granted though, a normal `pop()` on a `Vec` and a `VecDeque` is basically the same performance wise.
The difference is lies within a `pop_front` or `push_front` implementation.
Granted, I don't know if that means we also want to add a `Deque` trait for example.
Since then the question is, when does adding more and more traits become to cumbersome?

Reply:
Yeah that's what I was getting at in my other comment. I don't think it's going to be beneficial to has a new trait for each data structure type. That can probably be served somehow via `TypeData`.

But a `pop` method at least let's us handle lists, queues, and stacks in a general way.